frontend/src/features/visibility-settings/visibility-settings.slice.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { RootState } from '../../redux/reducer'
+import { getTable } from '../../data/table'
+
+interface VisibilitySettingsState {
```

Comment:
This settings is tightly coupled on the table and won't appear somewhere else. Please put the slicer under `TableState`

Visibility is also a vague word: row visibility? column visibility? Maybe add `column` or `variable` to provide more context.